### PR TITLE
CI: add MacOS to TravisCI build

### PR DESCRIPTION
**Pull request by mockdeep**

### CODE REVIEW

**Comment by mockdeep** [comment]
Weird, it doesn't seem to be building on MacOS. I'll have to debug this some more.

**Comment by Watson1978** [comment]
https://github.com/rmagick/rmagick/blob/b82717e6b333bfab0709a3185daffe87a4e45b5e/before_install_osx.sh#L2

I think we have to change above line to install ImageMagick 6 via Homebrew, like `brew install imagemagick@6`.

**Comment by mockdeep** [comment]
Yeah, the thing I'm confused about is the fact that the MacOS build didn't even run:

https://travis-ci.org/rmagick/rmagick/builds/494286894

I wonder if it's because of `sudo: required`.

File: .travis.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ cache:
 
 os:
   - linux
+  - macos
```
Comment (Watson1978) [inline .travis.yml]: I think you need to specify `osx` instead of `macos`.

https://docs.travis-ci.com/user/reference/osx/#using-macos

Comment (mockdeep) [reply .travis.yml]: doh, :sweat_smile:

**Comment by mockdeep** [comment]
Hey @Watson1978 I've got this building on MacOS, but it's failing for a lot of different reasons on different versions. [This one](https://travis-ci.org/rmagick/rmagick/jobs/536543632) has a `malloc` error, and [this one](https://travis-ci.org/rmagick/rmagick/jobs/536543655) has actual test failures. Do you have any ideas?

**Comment by Watson1978** [comment]
About SEGV, it reproduces on my environment with ruby 2.3.8 + IM 6.7.7 😣
https://gist.github.com/Watson1978/1429030ed27782e26d908bd247db1bb2

I will investigate it...

**Comment by Watson1978** [comment]
Seems that IM 6.7.7 causes heap buffer overflow and I guess it trigger SEGV.

When built RMagick with [AddressSanitizer](https://clang.llvm.org/docs/AddressSanitizer.html) and run test, it shows following logs.
https://gist.github.com/Watson1978/3184c749ca83fe85a8bd3b0c06f8a6b9

**Comment by mockdeep** [comment]
We're moving everything over to Github Actions.

**Pull request closed without merging**